src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2294,10 +2294,11 @@ else if (null != sourceCryptoMeta) {
                         } else {
                             String colValueStr = colValue.toString();
 
-                            // Remove extra trailing zeros added from toString
-                            if (colValue instanceof LocalDateTime || colValue instanceof LocalTime) {
-                                colValueStr = colValueStr.contains(".") ? colValueStr.replaceAll("0*$", "")
-                                                                        : colValueStr;
+                            if (colValue instanceof LocalDateTime) {
+                                colValueStr = ((LocalDateTime)colValue).format(DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss.SSSSSSS"));
+                            }
+                            else if (colValue instanceof LocalTime) {
+                                colValueStr = ((LocalTime)colValue).format(DateTimeFormatter.ofPattern("HH:mm:ss.SSSSSSS"));
                             }
 
                             if (unicodeConversionRequired(bulkJdbcType, destSSType)) {
```

Comment:
Also, since you are in here, the below references to colValue.toString() (lines 2304-2310) can be updated to colValueStr to avoid additional toString() calls.

Reply:
Done